Madness/Parser.swift

```diff
@@ -200,17 +207,21 @@ private func alternate<T, U>(left: Parser<T>.Function, right: Parser<U>.Function
 
 
 /// Defines repetition for use in the postfix `*` and `+` operator definitions above.
-private func repeat<T>(parser: Parser<T>.Function, _ interval: HalfOpenInterval<Int> = 0..<Int.max) -> Parser<[T]>.Function {
+private func repeat<T>(parser: Parser<T>.Function, _ interval: ClosedInterval<Int> = 0...Int.max) -> Parser<[T]>.Function {
 	if interval.end <= 0 { return { ([], $0) } }
-
+	
 	return { input in
 		parser(input).map { first, rest in
-			repeat(parser, (interval.start - 1)..<(interval.end - (interval.end == Int.max ? 0 : 1)))(rest).map {
+			repeat(parser, (interval.start - 1)...(interval.end - (interval.end == Int.max ? 0 : 1)))(rest).map {
 				([first] + $0, $1)
 			}
 		} ?? (interval.start <= 0 ? ([], input) : nil)
 	}
 }
+private func repeat<T>(parser: Parser<T>.Function, _ interval: HalfOpenInterval<Int> = 0..<Int.max) -> Parser<[T]>.Function {
```

Comment:
Could you copy the documentation over from the `ClosedInterval` overload, please?